### PR TITLE
[SOAR-7005] Update Teams Plugin with Troubleshooting from Discuss

### PR DESCRIPTION
**Pull request by jrose-r7**
## Proposed Changes
Add troubleshooting message to Teams
### Description

Describe the proposed changes:

  - Update the Teams plugin with troubleshooting information from Robert Helm on Discuss, specifically around how Teams will use the oldest team if there are two teams with the same name. https://discuss.rapid7.com/t/issuing-commands-from-teams-unsuccessful/6707/5?u=mike_rinehart

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `icon-plugin run -A -R all -T all`. For single action validation: `icon-plugin run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`icon-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

**Comment by jrose-r7** [comment]
May want to ignore UnitTest failure since the failure may be happening because it's looking for unit tests in the tests directory and not finding it even though they're there. @mrinehart-r7 is that a correct way to describe the issue?

File: plugins/microsoft_teams/help.md [hunk 1]
```diff
@@ -33,10 +33,7 @@ Example input:
   "application_id": "63a0cad6-ac64-435c-a221-5d37c97b763e",
   "application_secret": "aMeCAEYdOLlK+qRcD9AjdyxLkCaqZH1UPm7adjJQ5Og=",
   "directory_id": "9e538ff5-dcb2-46a9-9a28-f93b8250deb0",
-  "username_password": {
-    "username": "user",
-    "password": "mypassword"
-  }
+  "username_password": "{ \"username\": \"user\", \"password\": \"mypassword\" }"
```
Comment (pmara-r7) [inline plugins/microsoft_teams/help.md:36]: This is an icon-plugin generation problem and should be reverted

File: plugins/microsoft_teams/help.md [hunk 2]
```diff
@@ -195,7 +192,7 @@ Example input:
 ```
 {
   "channel_name": "InsightConnect Channel",
-  "message_content": "<b>Hello!</b>",
+  "message_content": "\u003cb\u003eHello!\u003c/b\u003e",
```
Comment (pmara-r7) [inline plugins/microsoft_teams/help.md:195]: Is this a good change? Not sure

**Review by pmara-r7 — CHANGES_REQUESTED** [review]
If you could fix the unit changes as well that would be nice but I guess its not part of this PR so up to you. If not, just reply that you don't want to fix rn and I can approve pending help.md changes

File: plugins/microsoft_teams/help.md (hunk 2 above)
Comment (mrinehart-r7) [reply plugins/microsoft_teams/help.md:195]: This should be reverted

File: plugins/microsoft_teams/help.md [hunk 3]
```diff
@@ -579,15 +576,15 @@ This action is used to create a group in Azure and enable it for Microsoft Teams
 |mail_enabled|boolean|None|True|Should e-mail should be enabled for this group|None|True|
 |mail_nickname|string|None|True|The nickname for the email address of this group in Outlook|None|TestGroup|
 |members|string[]|None|False|A list of usernames to set as members|None|["user@example.com"]|
-|owners|string[]|None|False|A list of usernames to set as owners|None|["user@example.com"]|
+[|owners|string[]|None|False|A list of usernames to set as owners|None|["user@example.com"]|
```
Comment (mrinehart-r7) [inline plugins/microsoft_teams/help.md:579]:
```suggestion
|owners|string[]|None|False|A list of usernames to set as owners|None|["user@example.com"]|
```

File: plugins/microsoft_teams/help.md [hunk 4]
```diff
@@ -869,10 +866,11 @@ Example output:
 
 ## Troubleshooting
 
-_This plugin does not contain any troubleshooting information._
+If there is more than one team with the same name in your organization, the oldest name between the two will be used.
```
Comment (mrinehart-r7) [inline plugins/microsoft_teams/help.md:869]:
```suggestion
If there is more than one team with the same name in your organization, the oldest team between the two will be used.
```

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

File: plugins/microsoft_teams/help.md (hunk 1 above)
Comment (jrose-r7) [reply plugins/microsoft_teams/help.md]: Thanks! reverted

File: plugins/microsoft_teams/help.md (hunk 2 above)
Comment (jrose-r7) [reply plugins/microsoft_teams/help.md]: Thanks! reverted

File: plugins/microsoft_teams/help.md [hunk 5]
```diff
@@ -33,9 +33,8 @@ Example input:
   "application_id": "63a0cad6-ac64-435c-a221-5d37c97b763e",
   "application_secret": "aMeCAEYdOLlK+qRcD9AjdyxLkCaqZH1UPm7adjJQ5Og=",
   "directory_id": "9e538ff5-dcb2-46a9-9a28-f93b8250deb0",
-  "username_password": {
-    "username": "user",
-    "password": "mypassword"
+  "username": "user",
+  "password": "mypassword"
   }
```
Comment (mrinehart-r7) [inline plugins/microsoft_teams/help.md:38]:
```suggestion
  "username_password": {
    "username": "user",
    "password": "mypassword"
  }
```

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

**Review by mrinehart-r7 — APPROVED** [review]

**Review by pmara-r7 — APPROVED** [review]

**Comment by pmara-r7** [comment]
Approved, discussion with @mrinehart-r7 said unit testing fix was out of scope of this ticket

**Pull request merged**